utils/detect_revision.py

```diff
@@ -44,14 +44,21 @@ def detect_debian_version():
 def detect_git_revision():
     try:
         cmd = subprocess.Popen(
-            ['git', 'rev-parse', '--short', 'HEAD'],
+            ['git', 'rev-parse', '--short=5', 'HEAD'],
             stdout=subprocess.PIPE,
             stderr=subprocess.PIPE, cwd=base_path
         )
         stdout, stderr = cmd.communicate()
         git_revnum = stdout.rstrip()
+        cmd = subprocess.Popen(
+            ['git', 'rev-parse', '--abbrev-ref', 'HEAD'],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE, cwd=base_path
+        )
+        stdout, stderr = cmd.communicate()
+        git_abbrev = stdout.rstrip()
         if git_revnum:
-            return 'unofficial-git-%s' % git_revnum
+            return 'git-%s[%s]' % (git_revnum, git_abbrev)
```

Comment:
Late to the party, but I'd like to repeat here again that the branch name is a local construct. Including it in the revision is meaningless outside of your local checkout. I think it is a mistake to use it in the revision identifiers, since it is not unique. Let's stick to number of commits (for lexical sorting) & sha for identifying.

Reply:
Although it is meaningless, the branch name will help us when low-tech people make bug reports on what they have been testing. If they have been testing a branch, we will automatically know that fro the version info.